### PR TITLE
Post to do and call streaks

### LINKED ISSUE #77: POST request for to do list
Create the POST request route to insert data into `Tasks` table.

Create a basic form for now with the following fields:
- title
- due date (use input type date)

### LINKED ISSUE #81: Update streaks on site load
Call the post streaks function to update the streaks every time the site loads

### PR DESCRIPTION
**Pull request by hritikbharucha**
## Tracking Info

Resolves #77 and Resolves #81

## Changes

- Created a SQL query to insert a new task with a title and date into the database
- Called the post streaks function on site load to always update streaks

## Testing

- Used Docker and LiveServer
- Console confirmed streaks were updated on site load and prevented duplicates
- Pop up confirmed task was successfully created

## Confirmation of Change

Screenshot 2024-06-04 at 9 55 00 PM

Screenshot 2024-06-04 at 9 58 03 PM

### CODE REVIEW

File: source/server.js [hunk 1]
```diff
@@ -13,10 +13,30 @@ const connection = mysql.createConnection({
     host: 'localhost',
     user: 'root',
     password: 'pw',
-    database: 'prod',
-    port: 3307
```
Comment (OasisAlex) [inline source/server.js:17]: Why is this deleted?

**Review by OasisAlex — CHANGES_REQUESTED** [review]
- Issue when I try to create a server, I think the issue was the missing line of port:3307
image
added it in my end, it's working after.
- [x] Try call streaks and view it in database 
image
image
image

- [x] POST tasks I used crul in command line 
image


image

File: source/server.js (hunk 1 above)
Comment (hritikbharucha) [reply source/server.js:17]: Oh good catch, I must have accidentally deleted at some point.

Comment (OasisAlex) [reply source/server.js:17]: Oh, okay. Then the rest looks fine

**Review by OasisAlex — APPROVED** [review]

**Review by EdwardLinS — CHANGES_REQUESTED** [review]
I tested both features and they work well. I changed the date of my computer to see if the streaks updated correctly and it works perfectly. 

Just one small request: in the issue, the task creation call should be called when the user submits through the form. Other than that one change, everything is good! Nice job!

**Comment by hritikbharucha** [comment]
> Just one small request: in the issue, the task creation call should be called when the user submits through the form. Other than that one change, everything is good! Nice job!

I added a temporary form in index.html that calls the task creation on form submission and confirmed that the tasks were added to the database. 
Screenshot 2024-06-06 at 11 24 31 PM

File: source/starter.sql [hunk 2]
```diff
@@ -7,7 +7,7 @@ CREATE TABLE Tasks (
     id INT AUTO_INCREMENT PRIMARY KEY,
     title VARCHAR(255) NOT NULL,
     due_date DATE,
-    completed BOOLEAN DEFAULT TRUE,
+    completed BOOLEAN DEFAULT FALSE,
```
Comment (EdwardLinS) [inline source/starter.sql:10]: For clarity, I changed this to allow inserted tasks to start as incompleted.

**Review by EdwardLinS — APPROVED** [review]
Thanks for adding the form! I looked through it all and everything works flawlessly. I re-added the port number in server for consistency with the rest of the repo and change it so that new inserted tasks start as incomplete through the starter.sql. 

Great job!

**Pull request merged**